lib/vm/src/trap/traphandlers.rs

```diff
@@ -1007,3 +1012,28 @@ pub fn lazy_per_thread_init() -> Result<(), Trap> {
         }
     }
 }
+
+extern "C" fn signal_less_trap_handler(pc: *const u8, trap: TrapCode) {
+    let jmp_buf = tls::with(|info| {
```

Comment:
Why do we need tls here?

Reply:
BTW, we already have the `wasmer_raise_trap` method.

Why not calling this one?
https://github.com/wasmerio/wasmer/blob/master/lib/vm/src/libcalls.rs#L670-L673

Reply:
Implemented this way to ensure we pass proper program counter. Can change, if you believe it makes sense.

Reply:
Can you expand on why do we need tls here @olonho?

Reply:
This is the location where jump buffer is stored and where we read trap info from. Not sure if properly understand the question. `raise_user_trap` for example is written in the same way.